cli/offchain-commands/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#[derive(Clone, Debug, clap::Subcommand)]
+#[allow(clippy::large_enum_variant)]
+pub enum OffchainCmd {
+	/// Get the script addresses for a given sidechain
+	Addresses,
+	/// Register a committee candidate
+	Register,
+	/// Deregister a committee candidate
+	Deregister,
+	/// Insert new D parameter
+	InsertDParameter,
```

Comment:
I think I will implement it as one, idempotent upsert-d-parameter command.

After it succeeded you can be sure, that the current value is what users wanted to be set. 

1) having two is not convenient for user
2) having two is not convenient from wizards developer perspective